scrapy/core/downloader/handlers/negotiation.py

```diff
@@ -183,29 +192,31 @@ def add_connection(self, transport, request: Request) -> None:
         negotiated_protocol = transport.negotiatedProtocol.decode()
         key = self._get_key(request)
         uri = self._parse_uri(request.url)
+        handler = self.handlers[negotiated_protocol]
 
         # Create an instance of the connection
         if negotiated_protocol == 'http/1.1':
+            assert isinstance(handler, HTTP11DownloadHandler)
```

Comment:
Is this a debug leftover? It seems it will fail if a user overwrites the handler for HTTP 1.1 with a custom handler that does not inherit from `HTTP11DownloadHandler`.